ooniprobe/Test/Suite/AbstractSuite.m

```diff
@@ -100,6 +100,13 @@ - (void)showLocalNotification {
 }
 
 -(NSArray*)getTestList {
+    if (self.autoRun) {
+        for (int i = 0; i < [self.testList count]; ++i) {
+            AbstractTest *current = self.testList[(NSUInteger) i];
+            current.autoRun = YES;
+            self.testList[(NSUInteger) i] = current;
+        }
+    }
```

Comment:
The current implementation needs to use state to pass information from the suite-specific `getTestList` to the abstract-suite's `getTestList`. It is unclean to use the state to pass information around like this, and we should see to refactor the code in a way that does not require us to use state here. Ideally, there should be a method that calls the sub-class or whatever `getTestList` and then code that filters and returns this list without using the state to pass information from the more specific class to the more generic class.